contracts/staking/SimpleStaking.sol

```diff
@@ -120,26 +120,28 @@ contract SimpleStaking is DSMath, Pausable, AbstractGoodStaking, StakingToken{
         if (tokenActual < tokenWithdraw) {
             tokenWithdraw = tokenActual;
         }
-        uint256 gdInterest =  InterestDistribution.withdrawStakeAndInterest(interestData, msg.sender, _amount);
+        //uint256 gdInterest =  InterestDistribution.withdrawStakeAndInterest(interestData, msg.sender, _amount);
         //Since we use generic ERC20 function we can just use its interface
-        ERC20 goodDollar = ERC20(address(avatar.nativeToken()));
-        fm.mintReward(msg.sender); // send rewards to user
+        //ERC20 goodDollar = ERC20(address(avatar.nativeToken()));
+        fm.mintReward(address(iToken), msg.sender); // send rewards to user
```

Comment:
move this as last action.
always perform token transfers last, after you update all the relevant data.
this prevents re-entry attacks

Reply:
moved